### PR TITLE
Remove `validate_numeric` and `validate_string`

### PR DESCRIPTION
**Pull request by alexjfisher**
Replace with data types

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -346,17 +346,17 @@
...
+  String[1] $template_snmptrapd           = $snmp::params::template_snmptrapd,
+  String[1] $template_snmptrapd_sysconfig = $snmp::params::template_snmptrapd_sysconfig,
   Boolean $openmanage_enable    = $snmp::params::openmanage_enable,
   Boolean $master               = $snmp::params::master,
   $agentx_perms                 = $snmp::params::agentx_perms,
   $agentx_ping_interval         = $snmp::params::agentx_ping_interval,
   $agentx_socket                = $snmp::params::agentx_socket,
-  $agentx_timeout               = $snmp::params::agentx_timeout,
-  $agentx_retries               = $snmp::params::agentx_retries,
+  Integer[0] $agentx_timeout    = $snmp::params::agentx_timeout,
```
Comment (alexjfisher) [inline manifests/init.pp]: Positive Integers only.

File: manifests/params.pp [hunk 2]
```diff
@@ -53,14 +53,14 @@
 
   $snmp_agentx_timeout = getvar('::snmp_agentx_timeout')
   if $snmp_agentx_timeout {
-    $agentx_timeout = $::snmp_agentx_timeout
+    $agentx_timeout = 0 + $::snmp_agentx_timeout
```
Comment (alexjfisher) [inline manifests/params.pp]: The rest of the params file says ENC vars are often strings and uses `str2bool` a lot.

File: manifests/init.pp [hunk 3]
```diff
@@ -346,17 +346,17 @@
   $trap_service_enable          = $snmp::params::trap_service_enable,
   $trap_service_hasstatus       = $snmp::params::trap_service_hasstatus,
   $trap_service_hasrestart      = $snmp::params::trap_service_hasrestart,
-  $template_snmpd_conf          = $snmp::params::template_snmpd_conf,
-  $template_snmpd_sysconfig     = $snmp::params::template_snmpd_sysconfig,
-  $template_snmptrapd           = $snmp::params::template_snmptrapd,
-  $template_snmptrapd_sysconfig = $snmp::params::template_snmptrapd_sysconfig,
+  String[1] $template_snmpd_conf          = $snmp::params::template_snmpd_conf,
```
[the remark this answers is not in the document]
Comment (alexjfisher) [reply manifests/init.pp]: meh.  I was in two minds.  See @ekohl's comment here. https://github.com/theforeman/puppet-pulp/pull/341#discussion_r220961197

Maybe realign it all at the end it you really want?  (ie after getting rid of the `validate_array`s)?

File: manifests/params.pp (hunk 2 above)
Comment (vStone) [reply manifests/params.pp]: Do we want to keep supporting top scoped (global) parameters? This could easily be solved in hiera?

Comment (ekohl) [reply manifests/params.pp]: Given it's a major version bump anyway, it's a good time to get rid of them IMHO

Comment (ekohl) [reply manifests/params.pp]: And Foreman can't set types for global parameters AFAIK., but can set them on class parameters in the ENC. Yet another reason to remove the global scoped parameters IMHO.

File: manifests/init.pp (hunk 3 above)
[the remark this answers is not in the document]
Comment (Dan33l) [reply manifests/init.pp]: We may align only the equal signs.

File: manifests/params.pp (hunk 2 above)
Comment (alexjfisher) [reply manifests/params.pp]: Maybe merge this for now and remove all the global parameter cruft in a separate PR?

Comment (ekohl) [reply manifests/params.pp]: That'd be https://github.com/voxpupuli/puppet-snmp/pull/145

**Pull request merged**